src/main/java/org/sagebionetworks/web/client/widget/entity/MarkdownWidgetViewImpl.java

```diff
@@ -41,28 +42,39 @@ public interface Binder extends UiBinder<Widget, MarkdownWidgetViewImpl> {}
   @UiField
   Italic emptyPanel;
 
+  private GWTWrapper gwt;
   public static GlobalApplicationState globalAppState;
   public static final EventListener relativeLinkClickHandler = event -> {
     event.preventDefault();
     if (Event.ONCLICK == event.getTypeInt()) {
       AnchorElement el = (AnchorElement) event.getCurrentTarget();
       String href = el.getHref();
-      String placeToken = href.substring(href.indexOf('!'));
-      AppPlaceHistoryMapper appPlaceHistoryMapper =
-        globalAppState.getAppPlaceHistoryMapper();
-      Place newPlace = appPlaceHistoryMapper.getPlace(placeToken);
-      globalAppState.getPlaceChanger().goTo(newPlace);
+      String placeToken = null;
+      // Detect old #!, and direct to the right place
+      if (href.contains("#!")) {
+        placeToken = href.substring(href.indexOf('!') + 1);
+      } else {
+        placeToken = StringUtils.getGWTPlaceTokenFromURL(href);
+      }
+      if (placeToken != null) {
+        AppPlaceHistoryMapper appPlaceHistoryMapper =
+          globalAppState.getAppPlaceHistoryMapper();
+        Place newPlace = appPlaceHistoryMapper.getPlace(placeToken);
+        globalAppState.getPlaceChanger().goTo(newPlace);
```

Comment:
And if you included a relative link without the '#!', it is optimized (internal SPA page change, not a page reload).